sklearn/utils/estimator_checks.py

```diff
@@ -1472,51 +1484,78 @@ def multioutput_estimator_convert_y_2d(name, y):
 
 
 @ignore_warnings(category=DeprecationWarning)
-def check_non_transformer_estimators_n_iter(name, estimator,
+def check_non_transformer_estimators_n_iter(name, Estimator,
                                             multi_output=False):
-    # Check if all iterative solvers, run for more than one iteration
+    # Test that estimators that are not transformers with an attribute max_iter,
+    # return the attribute of n_iter at least 1.
+
+    # These models are dependent on external solvers like
+    # libsvm and accessing the iter parameter is non-trivial.
+    not_run_check_n_iter = ['Ridge', 'SVR', 'NuSVR', 'NuSVC', 'RidgeClassifier',
+                            'SVC', 'RandomizedLasso', 'LogisticRegressionCV',
+                            'LinearSVC', 'LogisticRegression']
+
+    # Models made for multi-class scenarios
+    not_run_check_n_iter += ['MultiTaskElasticNet', 'MultiTaskElasticNetCV',
```

Comment:
These should work. There's a ``if multi_output`` below just for these. There's also a ``make_multioutput_y`` helper.